src/services/YotiSessionCompletionProcessor.ts

```diff
@@ -212,18 +212,29 @@ export class YotiSessionCompletionProcessor {
   				throw new AppError(HttpCodesEnum.SERVER_ERROR, "Missing Name Info in DocumentFields");
   			}
 
-  			if (missingGivenName || missingFamilyName) {
-  				const personDetails = await this.f2fService.getPersonIdentityById(f2fSession.sessionId, this.environmentVariables.personIdentityTableName());
+  			const personDetails = await this.f2fService.getPersonIdentityById(f2fSession.sessionId, this.environmentVariables.personIdentityTableName());
 
   				if (!personDetails) {
   					this.logger.warn("Missing details in PERSON IDENTITY tables", {
   						messageCode: MessageCodes.PERSON_NOT_FOUND,
   					});
   					throw new AppError(HttpCodesEnum.BAD_REQUEST, "Missing details in PERSON IDENTITY tables");
   				}
+				
+  			const { givenNames, familyNames } = personIdentityUtils.getNames(personDetails);
+  			const f2fGivenNames = givenNames.join(" ");
+  			const f2fFamilyName = familyNames.join(" ");
+		
+  			// Check if the fullName in f2f matches the fullName in DocumentFields
+  			const doesFullNameMatch = `${f2fGivenNames.toLowerCase()} ${f2fFamilyName.toLowerCase()}` === full_name.toLowerCase();
+  			if (!doesFullNameMatch) {
+  				this.logger.info("FullName mismatch between F2F & Yoti");
```

Comment:
this looks like it repeats line 22 in [src/utils/PersonIdentityUtils.ts](https://github.com/govuk-one-login/ipv-cri-f2f-api/pull/264/files#diff-1d9ffec70fa19ab3d3e30cd556a5ad663398e22efdb88fe5cb4ec7e488103f1d). Is there a reason we can't just return a server error here instead of even going into `getNamesFromPersonIdentity`?